Api 1281 configure event subs [DO NOT MERGE YET] (#839)
akeneo/pim-helpcenter: content/md/connectivity/manage-event-subscription.md

---
id: manage-event-subscription
themes: connectivity, import-export-data
title: Subscribe to **events**
popular: true
ee-only: false
related: what-is-the-event-subscription, how-to-connect-my-pim, manage-your-connections
---

::: info

> Not sure about what the event subscription is? You can find all the related information [right here](what-is-the-event-subscription.html).

Reviewer: ```suggestion
Not sure about what an event subscription is? You can find all the related information [right here](what-is-the-event-subscription.html).
```

Reviewer 2: ```suggestion
Before we jump right in, if you need to refresh your memory about [event subscriptions](what-is-the-event-subscription.html), take a look at our dedicated page.
```

::: 

> So, you're installing your first event-enabled extension and wondering how to make sure it gets everything it needs to work? 

Reviewer: ```suggestion
So, you are setting up your first event-enabled extension and you are wondering how to make sure it has everything it needs to work? 
```
Setting up instead of installing?

Reviewer 2: ```suggestion
You are about to set up your first event-enabled extension and you want to make sure it works. 
```

You've come to the right place. 

On this page, you'll find answers to the following questions:

> - How to enable the subscription?

Reviewer 2: ```suggestion
- How to enable an event subscription?
```

- How to configure the link with my extension?

# Activation

> First of all, the event subscription relies on connections, so you'll need to create a Connection if you haven't already. 

Reviewer: ```suggestion
First, an event subscription relies on the Akeneo PIM Connections feature, so you'll need to create a Connection first if you don't already have one.
```

> But don't worry, you will find everything you need on the following page: [Create a connection](manage-your-connections.html#create-a-connection). 

Reviewer: ```suggestion
The page [Create a connection](manage-your-connections.html#create-a-connection) will explain you how to create your first connection.
```

Reviewer 2: ```suggestion
You will find everything you need on the [Create a connection](manage-your-connections.html#create-a-connection) page. 
```

> While managing your extension, access the `Event Subscriptions` configuration page and use the toggle to turn it on. 

Reviewer 2: ```suggestion
In order to enable the feature, go to the `Event Subscriptions` configuration page, then use the toggle to turn it on.
```

![Using the Akeneo Events API](../img/event-subscription-helper.png)

> After a little more configuration, you'll be able to receive events.

Reviewer 2: ```suggestion
Now, let's move on to the URL configuration, and soon you'll be able to receive events.
```

# Configuration

## Request URL Configuration & Verification

> It's better if your Events Request URL is confirmed before saving the form. 

Reviewer: ```suggestion
It's better if your Event Request URL is confirmed before saving the form. 
```
Event instead of Events?

> To do so, please fill in the URL field with your server one, and click on the `Test` button to check if the URL is working. To help you, the Akeneo PIM displays the response code of the test request. 

Reviewer: ```suggestion
To do so, please fill in the URL field with your server URL, and click on the `Test` button to check if the URL is working. To help you, Akeneo PIM will display the response code of the test request. 
```

Reviewer 2: ```suggestion
To do so, please fill in the URL field with your server URL, and click on the `Test` button to check if the URL works. To help you, Akeneo PIM will display the response code of the test request. 
```

> Here are some examples of responses you may encounter. 

Reviewer: ```suggestion
Here are some examples of responses you may get. 
```

![URL Warnings](../img/url-warnings.png)

::: info

> If the third-party server takes some time to react and your initial attempt at URL verification fails due to a timeout, use the `Test` button to relaunch the verification.

Reviewer: ```suggestion
If the third-party server takes too much time to answer, your initial attempt at URL verification may fail due to a timeout, so use the `Test` button to relaunch the verification.
```

:::

## Manage your permissions

> The Event subscription leverages Akeneo PIM's permissions to control access to product data. 

Reviewer 2: ```suggestion
The Event subscription leverages Akeneo PIM permissions to control access to product data. 
```

> For example, suppose your extension needs to receive product update events only for your e-commerce products. In that case, you need to give your connection user group at least the `read` permission on the `e-commerce` category tree.

Reviewer: This example is not really clear, maybe you should specify that you need to give your connection user group at least the `read` permission on the category tree where your products are classified..
```suggestion
For example, suppose your extension needs to receive product update events only for your eCommerce products. In that case, you need to give your connection user group at least the `read` permission on the `eCommerce` category tree.
```

Reviewer 2: ```suggestion
For example, suppose your extension needs to receive product update events only for your eCommerce products. In that case, you need to give your connection user group at least the `read` permission on the `eCommerce` category tree, which is where your products are organized.
```

::: info

> You will find everything you need about connection user groups and permissions on the following page: [Manage your connections](manage-your-connections.html#configure-the-connection-user-group). 

Reviewer: ```suggestion
The [Manage your connections](manage-your-connections.html#configure-the-connection-user-group) article will give you more information about connection user groups and permissions.
```

Reviewer 2: ```suggestion
You will find everything you need about connection user groups and permissions on the [Manage your connections](manage-your-connections.html#configure-the-connection-user-group) page. 
```

:::

### More information about permissions

Let's take an example!

> 1. Your connection can view the English UK and French locales. 

Reviewer: ```suggestion
1. Your connection can view the English GB and French locales. 
```

> 2. Your connection has the read permission on the e-commerce category tree. 

Reviewer: ```suggestion
2. Your connection has the read permission on the eCommerce category tree. 
```

> 3. Your connection has the read permission on E-commerce, ERP, Marketing, and Media attribute groups.

Reviewer: ```suggestion
3. Your connection has the read permission on eCommerce, ERP, Marketing, and Media attribute groups.
```

When a product is created, updated, or deleted, you'll receive an event if: 

> - at least one product category is in the e-commerce category tree,

Reviewer: ```suggestion
- at least one product category is in the eCommerce category tree,
```

- an attribute value has been updated, regardless of its attribute group,
- an attribute value has been updated, regardless of the current locale.

> In the payload, you only receive attribute values that are in the E-commerce, ERP, Marketing, and Media attribute groups and for the English UK, and French locales.

Reviewer: ```suggestion
In the payload, you only receive attribute values that are in the eCommerce, ERP, Marketing, and Media attribute groups and for the English GB, and French locales.
```

::: warning
That means that you will receive an event if someone updates the German description of the product even if your connection is not allowed to see it. Same thing if someone updates an attribute that belongs to another attribute group.
:::

::: info

> If you are not comfortable with Akeneo PIM permissions, please find any information you need on the following page: [Set rights on your catalog](access-rights-on-products.html)

Reviewer: ```suggestion
If you are not comfortable with Akeneo PIM permissions, check the following article [Set rights on your catalog](access-rights-on-products.html)
```
